tests/components/mqtt/test_init.py

```diff
@@ -1112,40 +1112,50 @@ async def test_restore_all_active_subscriptions_on_reconnect(
     assert mqtt_client_mock.subscribe.mock_calls == expected
 
 
-async def test_setup_logs_error_if_no_connect_broker(hass, caplog):
+async def test_initial_setup_logs_error(hass, caplog, mqtt_client_mock):
+    """Test for setup failure if initial client connection fails."""
+    entry = MockConfigEntry(domain=mqtt.DOMAIN, data={mqtt.CONF_BROKER: "test-broker"})
+
+    mqtt_client_mock.connect.return_value = 1
+    assert await mqtt.async_setup_entry(hass, entry)
+    await hass.async_block_till_done()
+    assert "Failed to connect to MQTT server:" in caplog.text
+
+
+async def test_logs_error_if_no_connect_broker(hass, caplog, mqtt_mock):
     """Test for setup failure if connection to broker is missing."""
     entry = MockConfigEntry(domain=mqtt.DOMAIN, data={mqtt.CONF_BROKER: "test-broker"})
 
-    with patch("paho.mqtt.client.Client") as mock_client:
-        mock_client.on_connect(return_value=0)
-        assert await mqtt.async_setup_entry(hass, entry)
-        await hass.async_block_till_done()
-        assert "Failed to connect to MQTT server:" in caplog.text
-        component = hass.data["mqtt"]
-        component._mqtt_on_connect(mock_client, None, None, 3)
-        await hass.async_block_till_done()
-        assert "Unable to connect to the MQTT broker:" in caplog.text
+    assert await mqtt.async_setup_entry(hass, entry)
+    await hass.async_block_till_done()
+    # test with rc = 3 -> broker unavailable
+    mqtt_mock._mqtt_on_connect(mqtt_mock, None, None, 3)
```

Comment:
Can we use `mqtt_client_mock` instead of `mqtt_mock`?

Reply:
done